internal/devices/create.go

```diff
@@ -79,55 +78,51 @@ func (c *Client) Create() *cobra.Command {
 				userdata = string(userdataRaw)
 			}
 
+			deviceCreateInMetroInput := metal.NewDeviceCreateInMetroInput(metro, operatingSystem, plan)
+
+			var endDt time.Time
 			if terminationTime != "" {
 				parsedTime, err := time.Parse(time.RFC3339, terminationTime)
 				if err != nil {
 					return fmt.Errorf("Could not parse time %q: %w", terminationTime, err)
 				}
-				endDt = &packngo.Timestamp{Time: parsedTime}
-			}
-
-			var facilityArgs []string
-			if facility != "" {
-				facilityArgs = append(facilityArgs, facility)
+				endDt = parsedTime
 			}
-
-			request := &packngo.DeviceCreateRequest{
-				Hostname:              hostname,
-				Plan:                  plan,
-				Facility:              facilityArgs,
+			deviceCreateInMetroInput = &metal.DeviceCreateInMetroInput{
 				Metro:                 metro,
-				OS:                    operatingSystem,
-				BillingCycle:          billingCycle,
-				ProjectID:             projectID,
-				UserData:              userdata,
-				CustomData:            customdata,
-				IPXEScriptURL:         ipxescripturl,
+				Plan:                  plan,
+				OperatingSystem:       operatingSystem,
+				Hostname:              &hostname,
+				BillingCycle:          &billingCycle,
+				Userdata:              &userdata,
+				IpxeScriptUrl:         &ipxescripturl,
 				Tags:                  tags,
-				PublicIPv4SubnetSize:  publicIPv4SubnetSize,
-				AlwaysPXE:             alwaysPXE,
-				HardwareReservationID: hardwareReservationID,
-				SpotInstance:          spotInstance,
-				SpotPriceMax:          spotPriceMax,
-				TerminationTime:       endDt,
+				AlwaysPxe:             &alwaysPXE,
```

Comment:
AlwaysPxe now is a pointer in metal-go. In packngo there was no distinction between unset and zero values, so `false` (default value in line 143) was an empty value for `omitempty`, but now it is `false`. This causes the creation of a device with OS other than ipxe to fail in API since it shouldn't be specified.

I would rather check wether is empty and only include it depending on the `operatingSystem` field. Behavior for `ipxescripturl, hardwareReservationID, spotInstance, spotPriceMax` may need to be reviewed as well